mesonbuild/compilers/detect.py

```diff
@@ -573,6 +578,37 @@ def sanitize(p: str) -> str:
                 ccache, compiler, version, for_machine, is_cross, info,
                 exe_wrap, full_version=full_version, linker=linker)
 
+        if 'Metrowerks C/C++' in out or 'Freescale C/C++' in out:
+            lnk: T.Union[T.Type[linkers.MetrowerksLinkerARM], T.Type[linkers.MetrowerksLinkerEmbeddedPowerPC]]
```

Comment:
This variable has a different type up above, and due to Reasons, mypy considers it an error to change the type in a different branch. There are two general solutions to this:
- name it lnk2 or something
- raise its definition to be right underneath `cls: T.Union[T.Type[CCompiler], T.Type[CPPCompiler]]`; in this case however, you define it to match any StaticLinker or DynamicLinker, so it covers both Metrowerks and TI.